sklearn/decomposition/tests/test_incremental_pca.py

```diff
@@ -36,6 +38,32 @@ def test_incremental_pca():
                                   np.eye(X.shape[1]))
 
 
+def test_incremental_pca_sparse():
+    # Incremental PCA on sparse arrays.
+    X = iris.data
+    pca = PCA(n_components=2)
+    pca.fit_transform(X)
+    for cls in [sparse.csc_matrix, sparse.csr_matrix,
+                sparse.dok_matrix, sparse.lil_matrix]:
+        X = cls(X)
+        batch_size = X.shape[0] // 3
+        ipca = IncrementalPCA(n_components=2, batch_size=batch_size)
+
+        X_transformed = ipca.fit_transform(X)
+
+        np.testing.assert_equal(X_transformed.shape, (X.shape[0], 2))
+        assert_almost_equal(ipca.explained_variance_ratio_.sum(),
+                            pca.explained_variance_ratio_.sum(), 1)
```

Comment:
```suggestion
                            pca.explained_variance_ratio_.sum(), decimal=1)
```

Though I think we're switching to `np.testing.assert_allclose()`

Reply:
I just jumped to `assert_allclose` instead, putting in the smallest `atol` that would pass tests.